tasklist/importer-860/src/main/java/io/camunda/tasklist/zeebeimport/v860/processors/es/VariableZeebeRecordProcessorElasticSearch.java

```diff
@@ -102,4 +109,120 @@ private UpdateRequest getVariableQuery(final VariableEntity entity) throws Persi
           e);
     }
   }
+
+  private UpdateRequest persistVariableToListView(
+      final Record record, final VariableRecordValueImpl recordValue) throws PersistenceException {
+    final VariableEntity variableEntity = getVariableEntity(record, recordValue);
+    VariableListViewEntity variableListViewEntity = createVariableInputToListView(variableEntity);
+
+    if (isTaskOrSubProcessVariable(variableEntity)) {
+      variableListViewEntity = associateVariableWithTask(variableListViewEntity);
+      return prepareUpdateRequest(variableListViewEntity, variableListViewEntity.getScopeKey());
+    } else if (isProcessScope(variableEntity)) {
+      variableListViewEntity = associateVariableWithProcess(variableEntity, variableListViewEntity);
+      return prepareUpdateRequest(variableListViewEntity, variableEntity.getProcessInstanceId());
+    } else {
+      throw new PersistenceException(
+          String.format(
+              "Error to associate Variable with parent. Variable id: [%s]",
+              variableEntity.getId()));
+    }
+  }
+
+  private VariableListViewEntity createVariableInputToListView(final VariableEntity entity) {
+    final VariableListViewEntity variableListViewEntity = new VariableListViewEntity();
+    Optional.ofNullable(entity.getValue()).ifPresent(variableListViewEntity::setValue);
+    Optional.ofNullable(entity.getFullValue()).ifPresent(variableListViewEntity::setFullValue);
+    Optional.ofNullable(entity.getName()).ifPresent(variableListViewEntity::setName);
+    Optional.of(entity.getIsPreview()).ifPresent(variableListViewEntity::setIsPreview);
+    Optional.ofNullable(entity.getScopeFlowNodeId()).ifPresent(variableListViewEntity::setScopeKey);
+    Optional.ofNullable(entity.getId()).ifPresent(variableListViewEntity::setId);
+    Optional.of(entity.getPartitionId()).ifPresent(variableListViewEntity::setPartitionId);
+    Optional.ofNullable(entity.getTenantId()).ifPresent(variableListViewEntity::setTenantId);
+    variableListViewEntity.setJoin(new ListViewJoinRelation());
+
+    return variableListViewEntity;
+  }
+
+  private VariableListViewEntity associateVariableWithProcess(
+      final VariableEntity entity, final VariableListViewEntity variableListViewEntity) {
+    return associateVariableWithParent(
+        variableListViewEntity, "processVariable", entity.getProcessInstanceId());
+  }
+
+  private VariableListViewEntity associateVariableWithTask(
+      final VariableListViewEntity variableListViewEntity) {
+    return associateVariableWithParent(
+        variableListViewEntity, "taskVariable", variableListViewEntity.getScopeKey());
+  }
+
+  private VariableListViewEntity associateVariableWithParent(
+      final VariableListViewEntity variableListViewEntity,
+      final String name,
+      final String parentId) {
+    variableListViewEntity.getJoin().setName(name);
+    variableListViewEntity.getJoin().setParent(Long.valueOf(parentId));
+    return variableListViewEntity;
+  }
```

Comment:
I believe this can be improved for better readability and separation of concerns. The method `associateVariableWithParent` directly modifies the `VariableListViewEntity`, but it might be more intuitive to extract the logic for creating a `ListViewJoinRelation`. This will give us clearer inputs and outputs. 
```suggestion
  private ListViewJoinRelation createListViewJoinRelation(
      final String name,
      final String parentId) {
      final var result = new ListViewJoinRelation();
      result.setName(name);
      result.setParent(Long.valueOf(parentId));
      return result;
  }
```
same thing applies to associateVariableWithProcess and associateVariableWithTask